### ISSUE TITLE
Appropriate way listening to tag changes

### ISSUE THREAD

**Issue by yglodt** [PolymerEl/paper-tags#4]
What would be the appropriate way to listen for changes in the selected tags?

For every added/removed tag I want to execute a method, to update server-side.

**Comment by yglodt** [comment]
Note that listening for  iron-items-changed, iron-select, iron-deselect does not help in my case. I need to catch the moment someone is manually adding/removing a tag. 

iron-select, iron-deselect are also fired when I programmatically add/remove tags,

**Comment by christophe-g** [comment]
There are now a `tag-remove` and a `tag-added` event when tags are manually added or removed (fired by `paper-tags-input). Tell-me if this works for you.

Cheers,
C.

**Comment by yglodt** [comment]
Hi,

Thanks for your response and commits :)

tag-removed fires as expected. tag-added however not, at least in my app.

This is how I test:

```

        ready: function () {
            this.$.tagDropdown.addEventListener('iron-items-changed', function(e) {
                console.log("changed", e.type);
            }.bind(this));

            this.$.tagDropdown.addEventListener('iron-select', function(e) {
                console.log("select", e.type);
            }.bind(this));

            this.$.tagDropdown.addEventListener('iron-deselect', function(e) {
                console.log("deselect", e.type);
            }.bind(this));

            this.$.tagDropdown.addEventListener('tag-removed', function(e) {
                console.log("tag-removed", e.type);
            }.bind(this));

            this.$.tagDropdown.addEventListener('tag-added', function(e) {
                console.log("tag-added", e.type);
            }.bind(this));
        }
```

Note that iron-select still fires as expected on adding a tag.

**Comment by yglodt** [comment]
Here the more complete code example:

```
<dom-module id="object-relations">

    <template>

        <iron-ajax id="loadAllTags" 
            url="/api/tags" 
            handle-as="json" 
            last-response="{{tags}}">
        </iron-ajax>

        <iron-ajax id="loadTagsForObject"
            method="GET"
            url="/api/tagsforobject"
            handle-as="json" 
            last-response="{{valueObject}}"
            debounce-duration="1000">
        </iron-ajax>

        <iron-ajax id="saveTags"
            method="POST" 
            url="/api/tagsforobject" 
            content-type="application/json">
        </iron-ajax>

        <style>
            :host {
                display: block;
            }
        </style>

        <paper-tags-dropdown id="tagDropdown" items="[[tags]]" label="[[label]]" value-object="[[valueObject]]"></paper-tags-dropdown>
    </template>

    <script type="text/javascript">

    Polymer({
        is : "object-relations",
        properties: {
            label: {
                type: String,
                value: "Tags"
            }
        },
        ready: function () {
                    this.$.tagDropdown.addEventListener('tag-removed', function(e) {
                        console.log("tag-removed", e.type);
                    }.bind(this));

                    this.$.tagDropdown.addEventListener('tag-added', function(e) {
                        console.log("tag-added", e.type);
                    }.bind(this));
        }
    });
    </script>

</dom-module>
```

**Comment by renfeng** [comment]
I just played a bit with paper-tags-input. The property of items is not notified for change when adding tags. Thought, it works when removing tags.

**Comment by Stefdv** [comment]
In 'paper-tags-dropdown`  on the paper-menu-button use the correct 'on-iron-select' and 'on-iron-deselect'.
```
<paper-menu-button on-iron-select="_onSelection" on-iron-deselect="_onDeSelection">

Polymer({
...
 _onSelection:function(e,detail){
        this.fire('tag-added',detail.item._templateInstance.item)
      },
      _onDeSelection:function(e,detail){
         this.fire('tag-removed',detail.item._templateInstance.item)
      },
...
})

**Comment by christophe-g** [comment]
Solved by latest release (v1.7.2)
